src/test/java/app/controller/impl/ReviewControllerImplTest.java

```diff
@@ -0,0 +1,193 @@
+package app.controller.impl;
+
+import app.PopulatorTestUtil;
+import app.SecurityTestUtil;
+import app.config.AppConfig;
+import app.config.HibernateConfig;
+import app.dto.GameDTO;
+import app.dto.ReviewDTO;
+import app.entity.*;
+import app.mapper.GameMapper;
+import app.mapper.ReviewMapper;
+import io.javalin.Javalin;
+import io.javalin.http.HttpStatus;
+import io.restassured.RestAssured;
+import jakarta.persistence.EntityManagerFactory;
+import org.junit.jupiter.api.*;
+
+import java.util.List;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+
+class ReviewControllerImplTest {
+
+    private static Javalin app;
+    private static PopulatorTestUtil populatorTestUtil;
+    private static GameMapper gameMapper;
+    private static ReviewMapper reviewMapper;
+
+    private List<GameDTO> gameDTOList;
+    private List<ReviewDTO> reviewDTOList;
+
+    @BeforeAll
+    static void beforeAll() {
+        EntityManagerFactory emf = HibernateConfig.getEntityManagerFactoryForTest();
+        int port = 7070;
+
+        populatorTestUtil = new PopulatorTestUtil(emf);
+        app = AppConfig.startServer(port, emf);
+
+        gameMapper = new GameMapper();
+        reviewMapper = new ReviewMapper();
+
+        RestAssured.baseURI = String.format("http://localhost:%d/api", port);
+    }
+
+    @BeforeEach
+    void setUp() {
+        List<Role> roles = populatorTestUtil.createRoles();
+        populatorTestUtil.persist(roles);
+
+        List<User> users = populatorTestUtil.createUsers(roles);
+        populatorTestUtil.persist(users);
+
+        List<Genre> genres = populatorTestUtil.createGenres();
+        populatorTestUtil.persist(genres);
+
+        List<Platform> platforms = populatorTestUtil.createPlatforms();
+        populatorTestUtil.persist(platforms);
+
+        List<Game> games = populatorTestUtil.createGames(genres, platforms);
+        populatorTestUtil.persist(games);
```

Comment:
Persisting of games should maybe happen after the creation of reviews as creating reviews will set the `rating` and `ratingCount` attribute of the games they were made for.

It doesn't appear like games are ever queried or used again so it's probably not necessary for this test class.

Reply:
will not change since it doesn't improve anything